test/integration/scheduler/taint_test.go

```diff
@@ -666,7 +666,7 @@ func TestTaintBasedEvictions(t *testing.T) {
 
 			context = initTestScheduler(t, context, true, nil)
 			cs := context.clientSet
-			informers := context.informerFactory
+			informers := informers.NewSharedInformerFactory(context.clientSet, 0)
```

Comment:
Because in volume binding plugin, a pod informer is created on the scheduler informer factory (scheduler itself uses an independent pod informer with custom ListWatch), event handlers registered later by the controller will not take effect.

We should not share the informer factory between scheduler and controller.